service/history/shard/context_impl.go

```diff
@@ -2023,21 +2023,28 @@ func (s *ContextImpl) GetArchivalMetadata() archiver.ArchivalMetadata {
 	return s.archivalMetadata
 }
 
-func (s *ContextImpl) ensureMinContextTimeout(
+func (s *ContextImpl) newDetachedContext(
```

Comment:
comment about "in what situation you should used a detached Context"?

Reply:
https://github.com/temporalio/temporal/pull/3219